core/src/test/scala/org/http4s/EntityDecoderSpec.scala

```diff
@@ -37,6 +38,71 @@ class EntityDecoderSpec extends Http4sSpec with PendingUntilFixed {
         .run
         .run must be_-\/(ParseFailure("bummer", "real bummer"))
     }
+
+    val nonMatchingDecoder = EntityDecoder.decodeBy[String](MediaRange.`video/*`) { _ =>
+      DecodeResult.failure(ParseFailure("Nope.", ""))
+    }
+
+    val decoder1 = EntityDecoder.decodeBy(MediaType.`application/gnutar`) { msg =>
+      DecodeResult.success(1)
+    }
+
+    val decoder2 = EntityDecoder.decodeBy(MediaType.`application/excel`) { msg =>
+      DecodeResult.success(2)
+    }
+
+    val failDecoder = EntityDecoder.decodeBy[Int](MediaType.`application/soap+xml`) { msg =>
+      DecodeResult.failure(ParseFailure("Nope.", ""))
+    }
+
+    "Not match invalid media type" in {
+      nonMatchingDecoder.matchesMediaType(MediaType.`text/plain`) must_== false
+    }
+
+    "Match valid media range" in {
+      EntityDecoder.text.matchesMediaType(MediaType.`text/plain`) must_== true
+    }
+
+    "Match valid media type to a range" in {
+      EntityDecoder.text.matchesMediaType(MediaType.`text/css`) must_== true
+    }
+
+    "decodeStrict" >> {
+      "should produce a MediaTypeMissing if message has no content type" in {
+        val req = Request()
+        decoder1.decodeStrict(req) must_== -\/(MediaTypeMissing(decoder1.consumes))
+      }
+      "should produce a MediaTypeMismatch if message has unsupported content type" in {
+        val tpe = MediaType.`text/css`
+        val req = Request(headers = Headers(`Content-Type`(tpe)))
+        decoder1.decodeStrict(req) must_== -\/(MediaTypeMismatch(tpe, decoder1.consumes))
+      }
+    }
+
+    "composing EntityDecoders with orElse" >> {
+      "if a Message has no MediaType, the first one attempts to decode it" in { implicit ee: ExecutionEnv =>
+        val req = Request()
+        (decoder1 orElse decoder2).decode(req).run.run must_== DecodeResult.success(1).run.run
+        (failDecoder orElse decoder1).decode(req).run.run must_== DecodeResult.failure(ParseFailure("Nope.", "")).run.run
+      }
+      "A message with a MediaType that is not supported by any of the decoders" +
+        " will be attempted by the last decoder" in {
+        val reqMediaType = MediaType.`application/atom+xml`
+        val req = Request(headers = Headers(`Content-Type`(reqMediaType)))
+        val expected = DecodeResult.success(2)
+        (decoder1 orElse decoder2).decode(req).run.run must_== expected.run.run
+      }
+      "A catch all decoder will always attempt to decode a message" in {
```

Comment:
What are these supposed to test? By using `decode` as opposed to `decodeStrict` _any_ decoder will attempt to decode a message every time regardless of if it accepts `*/*`. Maybe I've missed the point of the test...